.travis.yml

```
@@ -13,56 +13,84 @@ notifications:
 cache:
   directories:
     - $HOME/.composer/cache
-    - $HOME/.phpbrew
+    - $HOME/.jest-cache
     - $HOME/.npm
-
-before_install:
-  - nvm install && nvm use
-  - npm install npm -g
+    - $HOME/.nvm/.cache
+    - $HOME/.phpbrew
+    - node_modules
 
 branches:
   only:
     - master
 
+before_install:
+  - nvm install
+
 jobs:
   include:
-    - stage: test
+    - name: JS unit tests
       env: WP_VERSION=latest
+      before_install:
+        - nvm install --latest-npm
+      install:
+        - npm install
```

```suggestion
        - npm ci
```

Rather than `install`, use `ci` as it is faster than `install`  when used with lock-files, and was created with CI services like Travis CI:

https://blog.npmjs.org/post/171556855892/introducing-npm-ci-for-faster-more-reliable

---

Yesterday in my testing I was seeing that caching `node_modules` and using `npm install` was about 20-30 seconds faster than using `npm ci` on our CI environment. Today, when double checking these results, I'm seeing that they're about the same 🤷‍♂️ 

My guess is that it all depends on what else is happening on the CI machine. If something is slowing down the network then `npm ci` will be faster. If something is slowing down the CPU then caching `node_modules` + `npm install` will be faster. If something is slowing down the disk then there's no escaping slow builds 😛

I'll switch to `npm ci` since it feels "more correct" but let's remember to keep re-evaluating this as our project evolves.

---

Sounds good to me 👍

Aside: npm cache details https://docs.npmjs.com/cli/cache#details